src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -939,23 +955,38 @@ private void AssignUnscheduledRequestsWithPlanByGreatestPlanTime(List<ScheduleRe
         {
             foreach (int idleNodeId in idleNodes)
             {
-                Dictionary<int, SchedulableRequest> configsWhichCanBeScheduledToThisNode = new Dictionary<int, SchedulableRequest>();
+                SchedulingPlan.PlanConfigData bestConfig = null;
+                SchedulableRequest bestRequest = null;
 
                 // Find the most expensive request in the plan to schedule from among the ones available.
                 foreach (SchedulableRequest request in _schedulingData.UnscheduledRequestsWhichCanBeScheduled)
                 {
                     if (CanScheduleRequestToNode(request, idleNodeId))
                     {
-                        configsWhichCanBeScheduledToThisNode[request.BuildRequest.ConfigurationId] = request;
+                        SchedulingPlan.PlanConfigData configToConsider = _schedulingPlan.GetConfiguration(request.BuildRequest.ConfigurationId);
+                        if (configToConsider is null)
+                        {
+                            if (bestConfig is null)
+                            {
+                                // By default we assume configs we don't know about aren't as important, and will only schedule them
+                                // if nothing else is suitable
+                                bestRequest = request;
```

Comment:
If we call `AssignUnscheduledRequestToNode` below, `bestConfig` must have been set to non-null. Isn't this assignment redundant? I.e. we seem to never use `bestRequest` unless we also set `bestConfig`.

Reply:
Same question about this logic in `AssignUnscheduledRequestsWithPlanByMostImmediateReferences`.

Reply:
Line 987 should say `if (bestRequest is not null)` instead of bestConfig, which makes this assignment relevant 🙂

It looks like I got it right in AssignUnscheduledRequestsWithPlanByMostImmediateReferences, but I'll fix it here. Thanks!